src/engine/app.rs

```diff
@@ -221,51 +217,70 @@ impl Application {
 }
 
 /// Helper builder for Applications.
-pub struct ApplicationBuilder<T>
+pub struct ApplicationBuilder<'a, T>
     where T: State + 'static
 {
     config: DisplayConfig,
     initial_state: T,
-    planner: Planner<()>,
+    dispatcher_builder: DispatcherBuilder<'a, 'a>,
+    world: World,
 }
 
-impl<T> ApplicationBuilder<T>
+impl<'a, T> ApplicationBuilder<'a, T>
     where T: State + 'static
 {
     /// Creates a new ApplicationBuilder with the given initial game state and
     /// display configuration.
-    pub fn new(initial_state: T, cfg: DisplayConfig) -> ApplicationBuilder<T> {
-        let pool = Arc::new(ThreadPool::new(num_cpus::get()));
+    pub fn new(initial_state: T, cfg: DisplayConfig) -> ApplicationBuilder<'a, T> {
+        use rayon::Configuration;
+
+        let pool = Arc::new(ThreadPool::new(Configuration::new().num_threads(num_cpus::get())).expect("Failed to create rayon::ThreadPool"));
 
         ApplicationBuilder {
             config: cfg,
             initial_state: initial_state,
-            planner: Planner::from_pool(World::new(), pool),
+            dispatcher_builder: DispatcherBuilder::new().with_pool(pool),
+            world: World::new(),
```

Comment:
Yeah I think so.  There is a space between when the Application is built and when we call `.run()` on it, so all it would take is a adding a `.world_mut()` function to the Application.